### ISSUE TITLE
angular-google-maps is unmaintained

### ISSUE THREAD

**Issue by jasonsreynolds** [freeCodeCamp/food-bank-app#19]
Implement alternative

**Comment by Rusticman** [comment]
Hi, I think I ought to tackle this as well if I'm going to get the route instructions as per the other issue.

**Comment by DealPete** [comment]
That would be great, this has been an issue for awhile.

**Comment by Rusticman** [comment]
I'm going to implement the alternative suggested on the ```angular-google-map``` github repo which is the very similarly named ```angularjs-google-maps```.

If you don't wish this to be the alternative then let me know. Here is a link to its github page:

https://github.com/allenhwkim/angularjs-google-maps

**Comment by DealPete** [comment]
Sounds good

**Comment by Rusticman** [comment]
Hey DealPete, 
                     thought I'd drop by to update you on progress. I had the alternative package ```angularjs-google-maps``` working in the same way as before except that it was duplicating the markers when you changed the state back and forth to the route assignment page. For a more detailed understanding, I've asked a question on SO. Here's the link:
http://stackoverflow.com/questions/41937470/combining-marker-clusterer-and-google-maps-api-within-angularjs-framework

TL:DR version of that question is; I've tried 4 separate solutions and come to the conclusion that it isn't possible to combine the ```marker clusterer``` package and the ```angularjs-google-maps``` package without creating duplicates. I await a response on SO however.

 I'm going to attempt using google maps api without having to use any package. Sorry about the delay!

**Comment by DealPete** [comment]
Hi Rusticman,

I'm not 100% sure I understand what's going on, but you have code in the controller to create markers when the user navigates to the page with the map. Controllers are instantiated every time you transition to a different state, so that would explain the duplicate markers. The map you receive with NgMap.getMap() must persist between calls to getMap. Would it work to create a map service that call getMap and creates the markers when the application is loaded? Each time the controller is instantiated it could query the service to get the current map.

**Comment by Rusticman** [comment]
I've used the google maps API now, and it was annoyingly easy to implement. I'm just learning how to squash the commits and I'll create a pull request. To further elaborate on the NgMap problem, read this message left by the owner:

https://github.com/allenhwkim/angularjs-google-maps/issues/575#issuecomment-210259209

 It explicitly requires the use of the <marker> directive. The marker clusterer package requires markers to be instantiated with the google maps Marker method in order for the clustering to work. I deleted the markers before the state changed, it didn't work.  I created a global variable array of marker objects in order to stop executing marker function, for example:
         `(array.length === 0){
createMarkers();
} ` 

It didn't matter, the markers would always come back as duplicates or the map would fail to render which is why the owner says,

 "All markers and shapes are destroyed and re-created when map is instantiated ONLY if users are using the given directives."

I'm happy to say that it now works without that package :)

**Comment by DealPete** [comment]
Fair enough, using our own code should make things simpler.
